.drone.yml

```diff
@@ -3,31 +3,91 @@ kind: pipeline
 name: pipeline1
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    restore: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+
 - name: checkstyle-and-sevntu
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh checkstyle-and-sevntu
 
 - name: spotbugs-and-pmd
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
```

Comment:
`.m2` folder in repository will be kept to the next steps (unlike `$USER/.m2`) - allows to untangle cache from build steps.